fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandlerSpec.groovy

```diff
@@ -0,0 +1,545 @@
+        1 * response.getFailureCallback(groupByQuery)
+
+        expect:
+        handler.handleRequest(context, request, groupByQuery, response)
+
+        and:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 1
+    }
+
+    def "Test handleRequest without building callback with json error"() {
+        setup:
+        final SuccessCallback success = Mock(SuccessCallback)
+        groupByQuery.getGranularity() >> DAY
+        WeightEvaluationQuery weightQuery = Mock(WeightEvaluationQuery)
+        queryWeightUtil.makeWeightEvaluationQuery(groupByQuery) >> weightQuery
+        writer.writeValueAsString(weightQuery) >> { throw new JsonProcessingException("word") }
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        ) {
+            @Override
+            public SuccessCallback buildCacheSuccessCallback(
+                    final SuccessCallback classicSuccessCallback,
+                    final CacheService cacheService,
+                    final DruidAggregationQuery<?> groupByQuery,
+                    final ResponseProcessor response
+            ) {
+                groupByQuery.clone()
+                return success
+            }
+        }
+
+        1 * queryWeightUtil.skipWeightCheckQuery(groupByQuery) >> false
+        0 * next.handleRequest(context, request, groupByQuery, response) >> true
+        1 * queryWeightUtil.getQueryWeightThreshold(DAY) >> 5
+        1 * groupByQuery.clone() >> groupByQuery
+        1 * webService.postDruidQuery(context, success, null, null, weightQuery)
+        1 * response.getErrorCallback(groupByQuery)
+        1 * response.getFailureCallback(groupByQuery)
+
+        expect:
+        handler.handleRequest(context, request, groupByQuery, response)
+
+        and:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 1
+    }
+
+    def "Test handle request on cache hit responds to the group by request"() {
+        setup:
+        GroupByQuery groupByQuery = RequestUtils.buildGroupByQuery()
+        QuerySigningService querySigningService = Mock(SegmentIntervalsHashIdGenerator)
+        querySigningService.getSegmentSetId(_) >> Optional.of(1234L)
+        ContainerRequestContext containerRequestContext = Mock(ContainerRequestContext)
+        containerRequestContext.getHeaders() >> (["Bard-Testing": "###BYPASS###", "ClientId": "UI"] as
+                MultivaluedHashMap<String, String>)
+        RequestContext requestContext = new RequestContext(containerRequestContext, true)
+        requestContext.isReadCache() >> true
+        MAPPER.valueToTree(groupByQuery) >> json
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
```

Comment:
If you mock the cacheService, you shouldn't need to mock these other containerRequest methods other than 'isReadCache'. ( I think this applies below as well)